ide/coqide/coqOps.ml

```diff
@@ -578,6 +609,19 @@ object(self)
           Minilib.log ("Unsupported feedback message " ^ mtype ^ " with a sentence")
       in
       let log ?id s = log_pp ?id (Pp.str s) in
+      (* When a command generates a warning or error, the message sent has an
+         incorrect sentence id.  If bp..ep is out of range for the specified
+         sentence, assume is it is from the last-sent "add" request *)
+      let handle_bad_loc loc id sentence =
+        match loc with
+        | Some loc ->
+          if loc.Loc.bp < sentence.bp || loc.Loc.ep > sentence.ep then begin
+              log_pp ?id Pp.(str "WarningMsg/ErrorMsg with bad loc, using tip instead");
```

Comment:
This should never happen in the first place, so instead of having this ugly workaround, make sure it doesn't ever happen.

Reply:
I don't know enough about Stm to know where to begin to fix it in there.